Add Money cell and Money type options UI

seancolsen opened the pull request:
Works towards #259

More details forthcoming


## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

pavish reviewed (changes requested):
@seancolsen @silentninja  I am not able to save a Money type column here.

1. When trying to convert a column to money type with empty values for decimal places and max digits, server throws error: the fields cannot be null.
2. When values are entered for both, server throws error: Unknown type_option passed.

seancolsen commented:
@pavish Yeah that's an issue that I brought up with @silentninja back when I was first working on this PR a couple months ago. My memory is hazy, but I vaguely recall there being an open ticket or open PR for that somewhere. I just went looking and couldn't find it though. @silentninja does this jog your memory? Do you know of a ticket or PR for this bug? If we don't have a ticket for that yet, then I'm happy to create one. But I don't think it needs to hold up this PR. @pavish you can continue your review by manually submitting the columns PATCH request without the `type_options` parameter. Then you can convert the column to Money and play with its behavior. You can even change its display options as long as you make sure to remove the `type_options` parameter from other PATCH requests to the columns endpoint too.

silentninja commented:
@pavish @seancolsen I think the issue is #1359

seancolsen commented:
Thanks @silentninja I'll make fixes in this PR. @pavish I'll ping you when it's ready for review.

seancolsen commented:
@pavish this is ready for re-review now.

- In 0cc393c31bc4e347084c8b41f2ec40a37ba40caa I removed the `DbTypeSelect` component as you recommended via chat. I also removed `DbOptionsForm`, placing its contents inline within `AbstractTypeOptions` because it was much simpler without `DbTypeSelect`.
- In ae9fcb4fff0481c7b03063fe0f0b370f7aae66ed I removed the SCALE and PRECISION DB type options.

pavish reviewed (approved):
@seancolsen Looks good to me.
